src/layers/microscopy-viewer-layer/data-utils/tiff-utils.js

```diff
@@ -1,50 +1,64 @@
+/* eslint-disable no-await-in-loop */
 // eslint-disable-next-line import/extensions
 import { fromUrl } from 'geotiff/dist/geotiff.bundle.min.js';
 
-async function loadTile({ image, channel, x, y, pool }) {
+async function getChannelConnections(channelUrl) {
+  const tiff = await fromUrl(channelUrl);
+  // Get the first image and check its size.
+  const pyramid = await tiff.parseFileDirectories();
+  const maxLevel = pyramid.length;
+  const pyramidLevels = [];
+  for (let i = 0; i < maxLevel; i += 1) {
+    const pyramidLevel = tiff.getImage(i);
+    pyramidLevels.push(pyramidLevel);
+  }
+  const resolvedConnections = await Promise.all(pyramidLevels);
+  return resolvedConnections;
+}
+
+async function loadTile({ image, x, y, pool }) {
```

Comment:
Oh yeah, most definitely. The code *within* the function needs to be executed synchronously. An `async` function returns a `Promise`, so in `loadTiff` we return an array of `loadTile` promises and resolve. If we didn't have `async` here, we couldn't await the response from `image.getTileOrStrip(x, y, 0, pool)` and there would be an error trying to decode the tile (since it would be a promise).

We should talk about `async/await` tomorrow as well. It's basically syntactic sugar for `promise.then()` but it is soo much easier to read and navigate logically IMO. I noticed some unnecessary/misused `Promise.all` from before.

Reply:
Apologies, I actually meant that on a literal level I hadn’t seen the `await` in the function so the `async` declaration seemed unnecessary - I think with the `await` on the `Promise.all` in the calling function, I was a little thrown off.  This stuff went has gone under a few changes so i was a little confused - I should have been clearer about that.